### PR TITLE
[IMP] mail: hide channel secret token earlier (welcome view)

### PR DESCRIPTION
**Pull request by tsm-odoo**

### CODE REVIEW

File: addons/mail/static/tests/tours/mail_channel_as_guest_tour.js [hunk 1]
```diff
@@ -6,6 +6,17 @@ import { createFile, inputFiles } from "web.test_utils_file";
 registry.category("web_tour.tours").add("mail/static/tests/tours/mail_channel_as_guest_tour.js", {
     test: true,
     steps: [
+        {
+            content: "Channel secret token has been hidden on welcome page",
+            trigger: ".o-mail-welcome-page",
+            run() {
+                if (!window.location.pathname.startsWith("/discuss/channel")) {
+                    if (!window.location.pathname.startsWith("/discuss/channel")) {
```
Comment (seb-odoo) [inline addons/mail/static/tests/tours/mail_channel_as_guest_tour.js]: double condition

File: addons/mail/static/src/new/public/discuss_public.js [hunk 2]
```diff
@@ -46,12 +43,23 @@ export class DiscussPublic extends Component {
                             }
                         });
                     }
+                } else if (isChannelTokenSecret) {
+                    this.removeChannelTokenFromURL();
```
Comment (seb-odoo) [inline addons/mail/static/src/new/public/discuss_public.js]: Do we still need this method to be present in useEffect, twice?
`useEffect` was used to detect change from welcome to channel, but in this case I feel like the change of URL could be done directly in the boot of the page, as soon as we have the necessary data to make the condition?

Comment (tsm-odoo) [reply addons/mail/static/src/new/public/discuss_public.js]: You're right, updated :+1:

File: addons/mail/static/src/new/public/discuss_public.js [hunk 3]
```diff
@@ -50,6 +42,12 @@ export class DiscussPublic extends Component {
             },
             () => [this.state.welcome]
         );
+        // Change the URL to avoid leaking the invitation link.
+        window.history.replaceState(
+            window.history.state,
+            null,
+            `/discuss/channel/${this.thread.id}${window.location.search}`
+        );
```
Comment (seb-odoo) [inline addons/mail/static/src/new/public/discuss_public.js]: Should only be done if `isChannelTokenSecret` is true

**Comment by seb-odoo** [comment]
there is conflict now

**Comment by tsm-odoo** [comment]
> there is conflict now

rebased

**Review by seb-odoo — APPROVED** [review]

**Pull request merged**